File: src/main/java/de/steklopod/sec01/Lec09AssignmentDemo.java
package de.steklopod.sec01;

import de.steklopod.courseutil.Util;
import de.steklopod.sec01.assignment.FileService;

public class Lec09AssignmentDemo {

    public static void main(String[] args) {

        FileService.write("file03.txt", "This is file3")
                .subscribe(Util.onNext(), Util.onError(), Util.onComplete());

        FileService.read("file03.txt")
                .subscribe(Util.onNext(), Util.onError(), Util.onComplete());

        FileService.delete("file03.txt")
                .subscribe(Util.onNext(), Util.onError(), Util.onComplete());

    }

}

File: src/main/kotlin/de/steklopod/sec01/Lec09AssignmentDemo.kt
package de.steklopod.sec01

import de.steklopod.utils.Util
import de.steklopod.sec01.assignment.FileService

object Lec09AssignmentDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        FileService.write("file03.txt", "This is file3")
            .subscribe(Util.onNext(), Util.onError(), Util.onComplete())
        FileService.read("file03.txt")
            .subscribe(Util.onNext(), Util.onError(), Util.onComplete())
        FileService.delete("file03.txt")
            .subscribe(Util.onNext(), Util.onError(), Util.onComplete())
    }
}
